in/kafkamdm/kafkamdm.go

```diff
@@ -62,108 +77,202 @@ func ConfigProcess(instance string) {
-	go k.notifications()
-	go k.consume()
-}
+	for _, topic := range topics {
+		// get partitions.
+		partitions, err := k.consumer.Partitions(topic)
+		if err != nil {
+			log.Fatal(4, "kafka-mdm: Faild to get partitions for topic %s. %s", topic, err)
+		}
+		for _, partition := range partitions {
+			switch offset {
+			case "oldest":
+				go k.consumePartition(topic, partition, -2)
+			case "newest":
+				go k.consumePartition(topic, partition, -1)
+			case "last":
+				o, err := offsetMgr.Last(topic, partition)
+				if err != nil {
+					log.Fatal(4, "kafka-mdm: Failed to get offset for %s:%d. %s", topic, partition, err)
+				}
+				go k.consumePartition(topic, partition, o)
+			default:
+				// treat offset as a time duration.
+				dur, err := time.ParseDuration(offset)
+				if err != nil {
+					log.Fatal(4, "kafka-mdm: Failed to parse offset to time.Duration. %s", err)
+				}
+				o, err := k.client.GetOffset(topic, partition, time.Now().Add(-1*dur).UnixNano()/int64(time.Millisecond))
+				if err != nil {
+					log.Fatal(4, "kafka-mdm: failed to get offset for %s:%d.  %s", topic, partition, err)
+				}
+				go k.consumePartition(topic, partition, o)
+			}
 
-func (k *KafkaMdm) consume() {
-	k.wg.Add(1)
-	messageChan := k.consumer.Messages()
-	for msg := range messageChan {
-		if LogLevel < 2 {
-			log.Debug("kafka-mdm received message: Topic %s, Partition: %d, Offset: %d, Key: %x", msg.Topic, msg.Partition, msg.Offset, msg.Key)
 		}
-		k.In.Handle(msg.Value)
-		k.consumer.MarkOffset(msg, "")
 	}
-	log.Info("kafka-mdm consumer ended.")
-	k.wg.Done()
 }
 
-func (k *KafkaMdm) notifications() {
+// this will continually consume from the topic until k.stopConsuming is triggered.
+func (k *KafkaMdm) consumePartition(topic string, partition int32, partitionOffset int64) {
 	k.wg.Add(1)
-	for msg := range k.consumer.Notifications() {
-		if len(msg.Claimed) > 0 {
-			for topic, partitions := range msg.Claimed {
-				log.Info("kafka-mdm consumer claimed %d partitions on topic: %s", len(partitions), topic)
-			}
-		}
-		if len(msg.Released) > 0 {
-			for topic, partitions := range msg.Released {
-				log.Info("kafka-mdm consumer released %d partitions on topic: %s", len(partitions), topic)
-			}
-		}
+	defer k.wg.Done()
 
-		if len(msg.Current) == 0 {
-			log.Info("kafka-mdm consumer is no longer consuming from any partitions.")
-		} else {
-			log.Info("kafka-mdm Current partitions:")
-			for topic, partitions := range msg.Current {
-				log.Info("kafka-mdm Current partitions: %s: %v", topic, partitions)
+	pc, err := k.consumer.ConsumePartition(topic, partition, partitionOffset)
+	if err != nil {
+		log.Fatal(4, "kafka-mdm: failed to start partitionConsumer for %s:%d. %s", topic, partition, err)
+	}
+	log.Info("kafka-mdm: consuming from %s:%d from offset %d", topic, partition, partitionOffset)
+	currentOffset := partitionOffset
+	messages := pc.Messages()
+	ticker := time.NewTicker(time.Second * 5)
```

Comment:
probably want to use offsetCommitInterval here. and maybe rename ticker to `offsetTicker`